documentation/docs/lab-share-lib.md

```diff
@@ -0,0 +1,116 @@
+# lab-share-lib
+
+`lab-share-lib` is a separate [library](https://github.com/sanger/lab-share-lib) that encapsulates AMQP connectivity logic, message deserialization and schema validation. Sections below are some discussions on how it encapsulates logic related to messaging infrastructure and schema validation.
+
+## Connectivity with Message Queues
+
+`tol-lab-share` establishes AMQP connections with a RabbitMQ broker. `lab-share-lib`, therefore, uses [`pika`](https://pika.readthedocs.io/en/stable/) library to manage connectivity logic and handle failures. `pika` uses several callback functions to handle messaging logic, and these are captured in [`lab_share_lib/rabbit/async_consumer.py`](https://github.com/sanger/lab-share-lib/blob/6724ec3c5053b75bb2a33958621930d9bb876a31/lab_share_lib/rabbit/async_consumer.py) in `lab-share-lib`.
+
+```python linenums="1" title="Snippets on connectivity logic and callbacks with RabbitMQ broker"
```

Comment:
I didn't know these were flags you could set on these code blocks.  Interesting.  Thanks.